pkg/apis/application/v1alpha1/types.go

```diff
@@ -446,7 +446,8 @@ type SyncSource struct {
 	// TargetBranch is the branch to which hydrated manifests should be committed
 	TargetBranch string `json:"targetBranch" protobuf:"bytes,1,name=targetBranch"`
 	// Path is a directory path within the git repository where hydrated manifests should be committed to and synced
-	// from. If hydrateTo is set, this is just the path from which hydrated manifests will be synced.
+	// from. The Path should never point to the root of the repo. If hydrateTo is set, this is just the path from which
+	// hydrated manifests will be synced.
 	Path string `json:"path" protobuf:"bytes,2,name=path"`
```

Comment:
Can we add kubebuilder validations to ensure 1) that the field is required, 2) that the minimum length is 1 character, and 3) that there's a regex rule preventing `.` and `/` (if possible, go regex is limited).